### PR TITLE
NLP implementation integration

### PR DESCRIPTION
**Pull request by KamenDimitrov97**
# What

In the latest pull request, improvements include enhanced response handling, added Berlin NLP location searching, introduced category-based NLP searching, improved search functionality with a scrubber, updated documentation for clarity, tidied up dependencies with 'go mod tidy,' and addressed a method name typo while removing unnecessary imports.
These changes aim to refine the project without introducing major disruptions.
Additionally, there were alterations to the Git history to verify old commits.

# How to review

Run all tests.
# Who can review

Anyone

### CODE REVIEW

File: api/search.go [hunk 1]
```diff
@@ -511,3 +540,89 @@ func processCountQuery(ctx context.Context, elasticSearchClient DpElasticSearche
...
+	// Process NLP Criteria based on the provided category data.
+	// If categories exist, iterate through them, limiting the loop based on the configuration
+	// NLP category limit. For each category, build NLP criteria to be used in the query to ElasticSearch.
+	if category != nil {
+		for i, cat := range *category {
+			if nlpSettings.CategoryLimit > 0 && nlpSettings.CategoryLimit <= i {
+				break
+			}
+			log.Info(ctx, cat.Code[0])
+			log.Info(ctx, cat.Code[1])
```
Comment (lindenmckenzie) [inline api/search.go:611]: This should be via `log.Data{}` with a more meaningful message.

File: query/query.go [hunk 2]
```diff
@@ -14,10 +14,30 @@ import (
...
+	UseCategory      bool
+	Categories       []NlpCriteriaCategory
+	UseSubdivision   bool
+	SubdivisionWords string
+}
+
+type NlpSettings struct {
+	CategoryWeighting float32 `json:"categoryWeighting"`
+	CategoryLimit     int     `json:"categoryLimit"`
+	DefaultState      string  `json:"defaultState"`
```
Comment (lindenmckenzie) [inline query/query.go:38]: prefer `snake_case` for json

File: swagger.yaml [hunk 3]
```diff
@@ -220,7 +220,208 @@ paths:
...
+        description: "The last health check date and time of the external service"
+        example: "2020-06-11T11:49:50.330089Z"
+      last_success:
+        type: string
+        description: "The last successful health check date and time of the external service"
+        example: "2020-06-11T11:49:50.330089Z"
+      last_failure:
+        type: string
+        description: "The last failed health check date and time of the external service"
+        example: "2019-09-22T11:48:51.0000001Z"
```
Comment (lindenmckenzie) [inline swagger.yaml:424]: Not sure why this has been added?

**Review by lindenmckenzie — COMMENTED** [review]
Some v minor comments

File: swagger.yaml (hunk 3 above)
Comment (KamenDimitrov97) [reply swagger.yaml]: I've forgotten to remove the definitions when we scrapped the nlp/search testing endpoint

**Review by lindenmckenzie — APPROVED** [review]

**Pull request merged**